src/Features/LanguageServer/Protocol/Protocol/CodeActionOptions.cs

```diff
@@ -35,11 +35,7 @@ internal class CodeActionOptions : IWorkDoneProgressOptions
         /// </summary>
         [DataMember(Name = "workDoneProgress")]
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
-        public bool WorkDoneProgress
-        {
-            get;
-            set;
-        }
+        public bool WorkDoneProgress { get; init; }
```

Comment:
Should we try to maintain alignment with the vslanguageserverclient versions of these files?

Reply:
I was thinking about this as well.  It's not strictly necessary since everything gets serialized, however it does make it easier to take changes from their side and apply them here.

On the other hand, it would be nice to modify these types so they're more ergonomic to use, and it shouldn't ever be extremely difficult to modify VS side changes into our style.  The typical changes are a couple properties being added or modified. 

There also may be other benefits of diverging - I expect we could switch to System.Text.Json earlier than the platform is switching for example.

So after a bit of a think, I'm OK with diverging.